### DISCUSSION TITLE
Docker lightweight image

### DISCUSSION THREAD

**Discussion by olegshulyakov** [open-webui/open-webui#6228] [Q&A]
Is there a way to get lightweight docker image of Open WebUI?

If it is used only as a front, running Ollama or OpenAI on a remote server, can docker image be smaller than 4GB? Like something about 100-200MB.

**Comment by nayefx64** [comment]
Yes, the image is getting bigger and bigger with each update for just a WebUI without models.

**Comment by alhimik45** [comment]
Any solution?

**Comment by pkeffect** [comment]
There are docker images for OI, OI with Ollama, OI Dev, OI and Ollama Dev, OI with CUDA AND OI Dev CUDA.

Obviously the OI image is the smallest. You also have the opportunity to install via venv, pyenv, uv and conda (miniconda) for native virtual environments.

**Reply by alhimik45** [reply]
I'm using version tag (0.6.4 now) without cuda/ollama postfix. So I guess that's what you are name smallest OI image. And it's still 4GB size

**Reply by olegshulyakov** [reply]
You cannot name 4GB image of web application as small. Web app is usually about 1-2MB =)

**Comment by fondoger** [comment]
I mainly use open-webui as a frontend app. The API requests are sent to cloud LLM providers. The docker image is too heavy right now.

**Comment by nayefx64** [comment]
The issue is with the size of how many Python Packages/Dependencies required by the Backend alone:

https://github.com/open-webui/open-webui/blob/5eca495d3e3b3066e7831141ed2adffbd6d179b4/backend/requirements.txt

###  Heavy Python Packages (Download + Installed Size)

| Package                    | Download Size | Installed Size | Notes |
|----------------------------|---------------|----------------|-------|
| `playwright`               | ~3 MB         | **300+ MB**    | Downloads browser binaries (~280 MB) |
| `chromadb`                 | ~4 MB         | **60–100 MB**  | Depends on `duckdb`, `uvicorn`, `clickhouse-driver`, etc. |
| `transformers`             | ~1 MB         | **150+ MB**    | Transformers lib only, no models |
| `sentence-transformers`    | ~300 KB       | **100–200 MB** | Pulls in `transformers`, `torch`, `scikit-learn` |
| `colbert-ai`               | ~1.5 MB       | **200+ MB**    | Includes `faiss`, `transformers`, `sentencepiece` |
| `opencv-python-headless`  | ~12 MB        | **100+ MB**    | Includes OpenCV libs |
| `unstructured`             | ~2 MB         | **100+ MB**    | Depends on `pdf2image`, `pytesseract`, etc. |
| `pandas`                   | ~10 MB        | **80+ MB**     | Includes `numpy`, `tzdata`, `python-dateutil` |
| `sqlalchemy`               | ~8 MB         | **20+ MB**     | ORM only |
| `psycopg2-binary`          | ~2 MB         | **10–15 MB**   | Binary Postgres driver |
| `aiohttp`                  | ~600 KB       | **12–15 MB**   | With `yarl`, `multidict`, etc. |

###  Total Estimated Size (Heaviest Packages)

| Metric               | Size        |
|----------------------|-------------|
| Total Download Size  | ~45–50 MB   |
| Total Installed Size | **1.2–1.6 GB** |

this is just half of them, the total is +3.5GB (only after pip install <package> for each)
If you look at the packages and why, it make sense because the UI holds a lot of features not to mention Playwrite and its own browser automation, also canvas rendering, Docs, TTS, RAG, Vector Store, etc. 

If you would install the OpenWebUI without Docker, you'll get the same size on your local environment/conda env.

I would treat this masterpiece as an operating system unless some lite UI comes out like OpenWebUILite or minimal use of it as LLM User Interface without fancy features.

A minimal version would be the same UI with less settings/options.

https://docs.openwebui.com/getting-started/advanced-topics/development/

**Reply by rgaricano** [reply]
plus node-modules, another big backpack!

**Reply by nayefx64** [reply]
Yes for the frontend
take this for example:
https://github.com/open-webui/open-webui/blob/5eca495d3e3b3066e7831141ed2adffbd6d179b4/package.json#L34
300MB+ in size, strictly for end-to-end testing.

If you dig deeper, you'll find more useful dev deps and less needed deps for runtime.

**Reply by yan12125** [reply]
Thanks for this great analysis. Unfortunately, dependencies are needed for features. For example, with `sentence-transformers` removed, basic conversations work but uploading files for analysis is broken.

**Comment by Jonas-Degn** [comment]
Here is a quick overview of the monthly size changes of the generated docker images since March 2025:

| Version| Main Size | Ollama Size | Cuda Size | 
|---|---|---|---|
| 0.6.25 | 5.57 GB | 8.31 GB | 11.60 GB |
| 0.6.18 | 4.78 GB | 7.58 GB| 10.70 GB |
| 0.6.15 | 4.84 GB | 8.22 GB | 10.80 GB |
| 0.6.10 | 4.60 GB | 7.96 GB | 10.60 GB |
| 0.6.5 | 4.59 GB | 7.97 GB | 8.91 GB |
| 0.6.0 | 4.50 GB | 7.87 GB | 8.82 GB |
| 0.5.20 | 4.49 GB | 7.68 GB | 8.57 GB |

- Base image has increase over 1 GB in size in the last 3 months
- Ollama fluctuates a lot more between versions but has also increased in size over the last 6 months
- Cuda image has increased massively in size over the last 4 months

## Large docker layer commands for main base image:
1.02 GB from "common system dependencies" from apt-get
0.92 GB from `pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cpu --no-cache-dir`
1.87 GB from `uv pip install --system -r requirements.txt --no-cache-dir`
0.45 GB in `./data/cache` from tiktoken, whisper and transformers
+ frontend, I didn't get around to check

## Common system dependencies for backend
| Package | Size | 
|---|---|
| TOTAL |  1.02 GB |
| git | 84 MB |
| build-essential | 240 MB |
| pandoc | 173 MB |
| python3-dev | 71 MB |
| ffmpeg | 450 MB |

## Python package distribution for 0.6.25 main
| Package | Size | 
|---|---|
| TOTAL |  2.7 GB |
| torch | 721 MB |
| milvus-lite | 169 MB |
| pyarrow | 135 MB |
| playwright  | 125 MB |
| tencentcloud | 109 MB |
| scipy | 98 MB |
| opencv_python.libs | 90 MB |
| googleapiclient | 86 MB |
| cv2 | 78 MB |
| sympy | 77 MB |
| ctranslate2.libs| 68 MB |
| av.libs | 64 MB |
| opencv_python_headless.libs | 62 MB |
| ctranslate2 | 62 MB |
| transformers | 58 MB |
| pandas | 53 MB |
| Packages under 50 MB each | 700-ish MB |

## Suggestion
Dockerfile - already uses multistage build to compile the frontend and then copies those final files into the backend image in a later layer.
**IF** any of large common system dependencies are **only** needed compile-time and not run-time, then I would advice building the backend in a multi-stage fashion just like the frontend. Pull the dependencies, compile what needs compilation, and copy the final files to a 3rd stage image where the compile-time system dependencies are not included.

Python packages - I don't know the purpose of all the packages, and so I don't know if any of the big packages should be hidden behind an `IF OLLAMA` or `IF CUDA` check, but it may - or may not - be worth considering in the community if a further minimized image is worth the development effort.

Very **very** early preliminary testing would indicate that, for me, to the extent I use open-webui, removing common dependencies `git build-essential pandoc gcc python3-dev` still provides a working open-webui.
I could suspect `pandoc` being needed for something run-time? but I suspect there is size to be saved by changing the dockerfile from 2 stages to 3 stages.

**Comment by olegshulyakov** [comment]
Do not care anymore, since I've made a [2MB version](https://github.com/olegshulyakov/llama.ui) for UI only 😂

**Reply by dmachinewhisperer** [reply]
thank you so much this is exactly what ive been looking for!.
